ui/app/hooks/useTimeout.js

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect, useRef, useCallback } from 'react'
+
+/**
+ * useTimeout
+ *
+ * @param {Function}  cb   - callback function inside setTimeout
+ * @param {number}  delay   - delay in ms
+ * @param {boolean}  immediate   - delay in ms
```

Comment:
This should be marked as optional, and the description should be updated. Something like this perhaps:
```suggestion
 * @param {boolean}  [immediate] - determines whether the timeout is invoked immediately
```